File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/MaxVolume.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class MaxVolume {
    
    /*@
    public normal_behavior
      ensures \result == ((x % 4294967296L + 6442450944L) % 4294967296L) - 2147483648L;
    model public static pure \bigint fb_wrap(\bigint x);

        public normal_behavior
          assignable \nothing;
          diverges s == Integer.MAX_VALUE;
          ensures s < Integer.MAX_VALUE;
          ensures \result >= 0;
          ensures (\forall int l, b; 1 <= l <= s && 1 <= b <= (\bigint)s - l + 1;
            fb_wrap((\bigint)l * b * ((\bigint)s - l - b)) <= \result);
          ensures \result == 0 || (\exists int l, b; 1 <= l <= s && 1 <= b <= (\bigint)s - l + 1;
            \result == fb_wrap((\bigint)l * b * ((\bigint)s - l - b)));
    @*/
    //@ code_java_math
    public static int maxVolume(int s) {
        int maxVolume = 0;

        /*@
        loop_invariant maxVolume >= 0;
        loop_invariant s < Integer.MAX_VALUE ==> 1 <= l && (\bigint)l <= (s < 1 ? 1 : (\bigint)s + 1);
        loop_invariant s < Integer.MAX_VALUE ==> (\forall int p, q; 1 <= p < l && 1 <= q <= (\bigint)s - p + 1; fb_wrap((\bigint)p * q * ((\bigint)s - p - q)) <= maxVolume);
        loop_invariant s < Integer.MAX_VALUE ==> maxVolume == 0 || (\exists int p, q; 1 <= p < l && 1 <= q <= (\bigint)s - p + 1; maxVolume == fb_wrap((\bigint)p * q * ((\bigint)s - p - q)));
        loop_writes l, maxVolume;
        @*/
        for (int l = 1; l <= s; l++) {
            /*@
            loop_invariant maxVolume >= 0;
            loop_invariant s < Integer.MAX_VALUE ==> 1 <= b <= (\bigint)s - l + 2;
            loop_invariant s < Integer.MAX_VALUE ==> (\forall int p, q; 1 <= p <= l && 1 <= q <= (\bigint)s - p + 1 && (p < l || q < b); fb_wrap((\bigint)p * q * ((\bigint)s - p - q)) <= maxVolume);
            loop_invariant s < Integer.MAX_VALUE ==> maxVolume == 0 || (\exists int p, q; 1 <= p <= l && 1 <= q <= (\bigint)s - p + 1 && (p < l || q < b); maxVolume == fb_wrap((\bigint)p * q * ((\bigint)s - p - q)));
            loop_writes b, maxVolume;
            @*/
            for (int b = 1; b <= s - l + 1; b++) {
                int h = s - l - b;
                int volume = l * b * h;
                if (volume > maxVolume) {
                    maxVolume = volume;
                }
            }
        }
        return maxVolume;
    }
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/MaxVolume.c
#include "java_arrays.h"
#include <stdint.h>

/*@
assigns \nothing;
ensures \result == (int32_t)((integer)left + right);
*/
static int32_t java_add(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left + (uint32_t)right);
}

/*@
assigns \nothing;
ensures \result == (int32_t)((integer)left - right);
*/
static int32_t java_sub(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left - (uint32_t)right);
}

/*@
assigns \nothing;
ensures \result == (int32_t)((integer)left * right);
*/
static int32_t java_mul(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left * (uint32_t)right);
}

/*@
terminates s < INT32_MAX;
assigns \nothing;
ensures s < INT32_MAX;
ensures \result >= 0;
ensures \forall integer l, b; 1 <= l <= s && 1 <= b <= s - l + 1 ==> (int32_t)(l * b * (s - l - b)) <= \result;
ensures \result == 0 || (\exists integer l, b; 1 <= l <= s && 1 <= b <= s - l + 1 && \result == (int32_t)(l * b * (s - l - b)));
*/
int32_t maxVolume(int32_t s) {
    int32_t maxVolume = 0;
    /*@
    loop invariant maxVolume >= 0;
    loop invariant s < INT32_MAX ==> 1 <= l <= (s < 1 ? 1 : (integer)s + 1);
    loop invariant s < INT32_MAX ==> (\forall integer p, q; 1 <= p < l && 1 <= q <= s - p + 1 ==> (int32_t)(p * q * (s - p - q)) <= maxVolume);
    loop invariant s < INT32_MAX ==> maxVolume == 0 || (\exists integer p, q; 1 <= p < l && 1 <= q <= s - p + 1 && maxVolume == (int32_t)(p * q * (s - p - q)));
    loop assigns l, maxVolume;
    */
    for (int32_t l = 1; l <= s; l = java_add(l, 1)) {
        int32_t upper = java_add(java_sub(s, l), 1);
        /*@
        loop invariant maxVolume >= 0;
        loop invariant s < INT32_MAX ==> 1 <= b <= (integer)upper + 1;
        loop invariant s < INT32_MAX ==> (\forall integer p, q; 1 <= p <= l && 1 <= q <= s - p + 1 && (p < l || q < b) ==> (int32_t)(p * q * (s - p - q)) <= maxVolume);
        loop invariant s < INT32_MAX ==> maxVolume == 0 || (\exists integer p, q; 1 <= p <= l && 1 <= q <= s - p + 1 && (p < l || q < b) && maxVolume == (int32_t)(p * q * (s - p - q)));
        loop assigns b, maxVolume;
        */
        for (int32_t b = 1; b <= upper; b = java_add(b, 1)) {
            int32_t h = java_sub(java_sub(s, l), b);
            int32_t volume = java_mul(java_mul(l, b), h);
            if (volume > maxVolume) {
                maxVolume = volume;
            }
        }
    }
    return maxVolume;
}
